### PR TITLE
make error messages more like serde

### PR DESCRIPTION
**Pull request by davidhewitt**

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -187,8 +206,17 @@ impl<'a> Parser<'a> {
...
+                // determine the error position to match serde
+                self.index += 1;
+                for c in NULL_REST.iter() {
+                    if !self.data.get(self.index + 1).is_some_and(|v| v == c) {
+                        break;
+                    }
+                    self.index += 1;
+                }
+                json_err!(ExpectedIdent, self.index)
+            }
```
Comment (davidhewitt) [inline src/parse.rs:219]: This change probably wants to be applied to `true` and `false` also

File: src/parse.rs [hunk 2]
```diff
@@ -79,16 +91,20 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub fn array_step(&mut self) -> JsonResult<bool> {
+    pub fn array_step(&mut self) -> JsonResult<Option<Peak>> {
```
Comment (davidhewitt) [inline src/parse.rs:94]: I didn't have a chance to look at performance implications of this change, it seemed necessary in order to support trailing comma as an error.

File: src/parse.rs (hunk 1 above)
Comment (samuelcolvin) [reply src/parse.rs]: done.

File: src/parse.rs (hunk 2 above)
Comment (samuelcolvin) [reply src/parse.rs:95]: it's good.

**Pull request merged**